### PR TITLE
ORCA makefile improvements

### PR DESCRIPTION
**Pull request by hlinnaka**

### CODE REVIEW

File: src/backend/Makefile [hunk 1]
```diff
@@ -67,14 +67,19 @@ LIBS := $(filter-out -lpgport, $(LIBS)) $(LDAP_LIBS_BE) $(CONNECTEMCLIB)
...
+LIBS := $(LIBS) -L$(LIBGPOS)/$(OBJDIR_DEFAULT)
+LIBS := $(LIBS) -L$(OPTIMIZER)/libnaucrates/$(OBJDIR_DEFAULT)
+LIBS := $(LIBS) -L$(OPTIMIZER)/libgpdbcost/$(OBJDIR_DEFAULT)
+LIBS := $(LIBS) -L$(OPTIMIZER)/libgpopt/$(OBJDIR_DEFAULT)
+LIBS := $(LIBS) -L$(XERCES_LIBDIR)
+
+# The order of the libraries is significant, when building against static
+# ORCA libraries.
+LIBS := $(LIBS) -lgpopt -lnaucrates -lgpdbcost -lgpos
+LIBS := $(LIBS) -lxerces-c-3.1
```
Comment (danielgustafsson) [inline src/backend/Makefile]: This seems to drop the dxltranslators lib, isn't that required? On that note, to make the build less bespoke I still think we should install that lib like any other as I tried in #428 : https://github.com/greenplum-db/gpdb/pull/428/commits/c09e36a0de3a0e89021dff83d2978fc3aa4deb02#diff-2ecbac720118e8f0744f98a785fd7084

Comment (hlinnaka) [inline src/backend/Makefile]: On 07/15/2016 09:26 AM, Daniel Gustafsson wrote:
>> -LIBS := $(LIBS) -L$(XERCES_LIBDIR) -lxerces-c-3.1
>> -LIBS := $(LIBS) -L$(OPTIMIZER)/libnaucrates/$(OBJDIR_DEFAULT) -lnaucrates
>> -LIBS := $(LIBS) -L$(OPTIMIZER)/libgpdbcost/$(OBJDIR_DEFAULT) -lgpdbcost
>> -LIBS := $(LIBS) -L$(OPTIMIZER)/libgpopt/$(OBJDIR_DEFAULT) -lgpopt
>> -LIBS := $(LIBS) -L$(top_builddir)/src/backend/gpopt -ldxltranslators
>> +
>> +LIBS := $(LIBS) -L$(LIBGPOS)/$(OBJDIR_DEFAULT)
>> +LIBS := $(LIBS) -L$(OPTIMIZER)/libnaucrates/$(OBJDIR_DEFAULT)
>> +LIBS := $(LIBS) -L$(OPTIMIZER)/libgpdbcost/$(OBJDIR_DEFAULT)
>> +LIBS := $(LIBS) -L$(OPTIMIZER)/libgpopt/$(OBJDIR_DEFAULT)
>> +LIBS := $(LIBS) -L$(XERCES_LIBDIR)
>> +
>> +# The order of the libraries is significant, when building against static
>> +# ORCA libraries.
>> +LIBS := $(LIBS) -lgpopt -lnaucrates -lgpdbcost -lgpos
>> +LIBS := $(LIBS) -lxerces-c-3.1
>
> This seems to drop the dxltranslators lib, isn't that required? On that note, to make the build less bespoke I still think we should install that lib like any other as I tried in #428 : https://github.com/greenplum-db/gpdb/pull/428/commits/c09e36a0de3a0e89021dff83d2978fc3aa4deb02#diff-2ecbac720118e8f0744f98a785fd7084

Ah, good catch. My ultimate goal is actually to get rid of 
libdxltranslators.so as a separate library altogether. Instead, link all 
the src/backend/gpopt/ stuff directly into the postgres binary, like any 
other subdirectory. There's no real reason for that to be a separate 
library.

That's why I accidentally removed -ldxltranslators from the above: I 
extracted these Makefile changes from a larger patch I was hacking on, 
which removed libdxltranslators.so altogether.

I have updated this PR with a new set of patches. It now includes the 
change to remove libdxltranslators.so. Turned out to be simpler to 
commit that change first.

- Heikki

Comment (danielgustafsson) [inline src/backend/Makefile]: Fair enough, I don't object to that at all.

**Comment by hsyuan** [comment]
LGTM. It is good to get rid of dxl translators lib.

**Comment by hlinnaka** [comment]
Pushed, thanks!

**Pull request closed without merging**